paddle/trainer/TrainerConfigHelper.cpp

```diff
@@ -36,6 +36,22 @@ namespace paddle {
 
 struct TrainerConfigHelperPrivate {
   TrainerConfig conf;
+
+  /**
+   * @brief Update trainer default values to each Parameter.
+   */
+  void updateDefaultValues() {
+    // set default gradient clipping.
+    if (conf.default_values().has_gradient_clipping_threshold()) {
+      auto defaultGradientClipping =
+          conf.default_values().gradient_clipping_threshold();
+      for (auto &param : *conf.mutable_model_config()->mutable_parameters()) {
+        if (!param.has_gradient_clipping_threshold()) {
+          param.set_gradient_clipping_threshold(defaultGradientClipping);
+        }
+      }
+    }
+  }
```

Comment:
In this PR, only `gradient_clipping_threshold` will be fixed. This PR is trying to explore a unified way to fix global variables bugs, not trying to fix them in one PR.

The reset variables would be fixed by following PRs.